src/App.tsx

```diff
@@ -21,55 +20,152 @@ enum SortType {
   LENGTH,
 }
 
-// Use this function in the render method
-function getReorderedGoods(
+type State = {
   goods: string[],
-  sortType: SortType,
+  isVisible: boolean,
   isReversed: boolean,
-) {
-  // Not to mutate the original array
-  const visibleGoods = [...goods];
+  isSorted: boolean,
+  sortBy: SortType,
+};
 
-  // Sort and reverse goods if needed
-  // ...
+class App extends React.Component<{}, State> {
+  state:Readonly<State> = {
```

Comment:
You've already set your type in generic
```suggestion
  state = {
```